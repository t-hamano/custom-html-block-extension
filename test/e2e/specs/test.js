/**
 * WordPress dependencies
 */
import {
	insertBlock,
	installPlugin,
	activatePlugin,
	deactivatePlugin,
	uninstallPlugin,
	pressKeyWithModifier,
	switchUserToAdmin,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	dismissPointer,
	clickButtonWithId,
	createNewClassicPost,
	clickAdminTopMenuWithText,
	expandAdminMenu,
	clickAdminSubMenuWithText,
} from '../helper';

const page = global.page;

page.on( 'dialog', async ( dialog ) => await dialog.accept() );

describe( 'Editor', () => {
	beforeAll( async () => {
		await switchUserToAdmin();
		await dismissPointer();
	} );

	it( 'Input by Emmet should be expanded on classic editor', async () => {
		await installPlugin( 'classic-editor' );
		await activatePlugin( 'classic-editor' );
		await createNewClassicPost();
		await clickButtonWithId( 'content-tmce' );
		await clickButtonWithId( 'content-html' );
		await page.waitForSelector( '#monaco-editor-container .monaco-editor' );
		await page.click( '#monaco-editor-container .monaco-editor' );
		await page.keyboard.type( 'p.selector' );
		await page.keyboard.down( 'Tab' );
		await page.waitForTimeout( 500 );
		await clickButtonWithId( 'publish' );
		await clickButtonWithId( 'content-tmce' );
		await clickButtonWithId( 'content-html' );
		await page.waitForSelector( '#wp-content-editor-container textarea.wp-editor-area' );
		await page.waitForTimeout( 1000 );
		const textarea = await page.$$( '#wp-content-editor-container textarea.wp-editor-area' );
		const textareaValue = await page.evaluate( ( element ) => element.textContent, textarea[ 0 ] );
		expect( textareaValue ).toBe( '<p class="selector"></p>' );
		await deactivatePlugin( 'classic-editor' );
		await uninstallPlugin( 'classic-editor' );
	} );

	it( 'Input by Emmet should be expanded on theme editor', async () => {
		await clickAdminTopMenuWithText( 'Appearance' );
		await expandAdminMenu();
		await clickAdminSubMenuWithText( 'Theme Editor' );
		await page.waitForSelector( '#monaco-editor-container .monaco-editor' );
		await page.click( '#monaco-editor-container .monaco-editor' );
		await pressKeyWithModifier( 'primary', 'a' );
		await page.keyboard.press( 'Delete' );
		await page.keyboard.type( '.selector{fz100', { delay: 50 } );
		await page.keyboard.press( 'Tab' );
		await page.waitForTimeout( 500 );
		await clickButtonWithId( 'submit' );
		await page.waitForSelector( '#monaco-editor-container .monaco-editor' );
		await page.waitForSelector( '#template textarea' );
		const textarea = await page.$$( '#template textarea' );
		const textareaValue = await page.evaluate( ( element ) => element.textContent, textarea[ 0 ] );
		expect( textareaValue ).toBe( '.selector{font-size: 100px;}' );
	} );

	it( 'Input by Emmet should be expanded on block editor', async () => {
		await createNewPost();
		await insertBlock( 'Custom HTML' );
		await page.waitForSelector( '[data-type="core/html"] .monaco-editor' );
		await page.click( '[data-type="core/html"] .monaco-editor' );
		await page.keyboard.type( 'ul.list>li.item*5' );
		await page.keyboard.down( 'Tab' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
