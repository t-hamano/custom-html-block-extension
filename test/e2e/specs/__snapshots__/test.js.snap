// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor input by Emmet should be expanded on block editor 1`] = `
"<!-- wp:html -->
<ul class="list">
  <li class="item"></li>
  <li class="item"></li>
  <li class="item"></li>
  <li class="item"></li>
  <li class="item"></li>
</ul>
<!-- /wp:html -->"
`;
